### PR TITLE
[SageMaker] Move SageMaker partition job to use SageMaker Processing instead of Training

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

*Description of changes:*

* Convert the SageMaker partition jobs from training to processing.
* Add Docker login to SM Docker build script
* Add and pin dependencies to SM Docker container

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docker/sagemaker/Dockerfile.sm [hunk 1]
```diff
@@ -18,7 +18,15 @@ FROM branch-${DEVICE} AS final
 LABEL maintainer="Amazon AI Graph ML team"
 
 # Install related Python packages
-RUN pip3 install ogb==1.3.6 scipy pyarrow boto3 scikit-learn transformers==4.28.1 \
+RUN pip3 install \
+        boto3 \
+        numba==0.58.1 \
+        numpy==1.26.1 \
```
Comment (classicsong) [inline docker/sagemaker/Dockerfile.sm:24]: Does SageMaker partition rely on specify numpy and numba version?

File: python/graphstorm/sagemaker/sagemaker_partition.py [hunk 2]
```diff
@@ -214,7 +214,8 @@ def run_partition(job_config: PartitionJobConfig):
     metadata_filename = job_config.metadata_filename
     skip_partitioning = job_config.skip_partitioning == 'true'
 
-    sm_env = json.loads(os.environ['SM_TRAINING_ENV'])
+    with open("/opt/ml/config/resourceconfig.json", "r", encoding="utf-8") as f:
```
Comment (classicsong) [inline python/graphstorm/sagemaker/sagemaker_partition.py:217]: Is SageMaker API changed?

File: sagemaker/launch/launch_partition.py [hunk 3]
```diff
@@ -45,40 +44,43 @@ def run_job(input_args, image):
...
-    est = PyTorch(
-        disable_profiler=True,
-        debugger_hook_config=False,
-        entry_point=os.path.basename(entry_point),
-        source_dir=os.path.dirname(entry_point),
+    script_processor = ScriptProcessor(
         image_uri=image,
         role=role,
-        instance_count=instance_count,
+        instance_count=int(instance_count),
```
Comment (classicsong) [inline sagemaker/launch/launch_partition.py:67]: Shouldn't instance_count be a int? We can define the type of --instance-count

**Review by classicsong — APPROVED** [review]

File: docker/sagemaker/Dockerfile.sm (hunk 1 above)
Comment (thvasilo) [reply docker/sagemaker/Dockerfile.sm:24]: I've encountered this particular error when not pinning versions: https://github.com/numba/numba/issues/7339

File: python/graphstorm/sagemaker/sagemaker_partition.py (hunk 2 above)
Comment (thvasilo) [reply python/graphstorm/sagemaker/sagemaker_partition.py:217]: No the launch calls remain the same from the user's perspective. The information that exists as env vars in Training exists in this file in Processing jobs.

File: sagemaker/launch/launch_partition.py (hunk 3 above)
Comment (thvasilo) [reply sagemaker/launch/launch_partition.py:67]: Yeah actually that's already the case, so I'll remove this : https://github.com/awslabs/graphstorm/blob/main/sagemaker/launch/common_parser.py#L33

**Pull request merged**